### ISSUE TITLE
request method ANY 

### ISSUE THREAD

**Issue by theburhan** [spring-cloud/spring-cloud-contract#1039]
Enabling Stub Runner adds default stubs /health and /ping, I am getting an additional stub which mess up with any other stubs that I have defined as it always matches with the one mentioned below.
```
{
  "id" : "35a1d3cc-e164-4ddc-a535-e3f267c9a9d8",
  "request" : {
    "method" : "ANY"
  },
  "response" : {
    "status" : 200
  },
  "uuid" : "35a1d3cc-e164-4ddc-a535-e3f267c9a9d8"
}
```

Any leads to that, why is that generated, whereas in DSL contracts I have not defined anything similar.

**Comment by marcingrzejszczak** [comment]
Hi! I don't understand the problem. Can you rephrase it to state how you think that the framework should behave?

**Comment by theburhan** [comment]
Hi @marcingrzejszczak  I have a spring boot application with  annotation @EnableStubRunnerServer and with configuration 
```
stubrunner:
  ids:
    - com.company.service:application:+:stubs
```

When I run the application, then stubs are loaded with class path, with three additional stubs `/health`, `ping` that you have mentioned in an earlier post. I am getting another one with request method `ANY`.  Here are the logs.


```
{
  "id" : "741a51c5-02d5-4a48-91f5-f86438639615",
  "request" : {
    "url" : "/ping",
    "method" : "GET"
  },
  "response" : {
    "status" : 200,
    "body" : "OK"
  },
  "uuid" : "741a51c5-02d5-4a48-91f5-f86438639615"
}

WireMock                                 : Received request to /mappings with body {
  "id" : "e712b70e-36da-4a05-8de5-ce89dfaf6baf",
  "request" : {
    "url" : "/health",
    "method" : "GET"
  },
  "response" : {
    "status" : 200,
    "body" : "OK"
  },
  "uuid" : "e712b70e-36da-4a05-8de5-ce89dfaf6baf"
}

 WireMock                                 : Received request to /mappings with body {
  "id" : "3c192245-23cc-4292-bbd9-2e9275ff5435",
  "request" : {
    "method" : "ANY"
  },
  "response" : {
    "status" : 200
  },
  "uuid" : "3c192245-23cc-4292-bbd9-2e9275ff5435"
}

WireMock                                 : Received request to /mappings with body {
  "id" : "0a72710f-39d8-4beb-be53-358bb5563c57",
  "request" : {
    "url" : "/v1/application",
    "method" : "POST",
    "headers" : {
      "Content-Type" : {
        "matches" : "application/json.*"
      },
      "Accept" : {
        "matches" : "application/hal\\+json.*"
      },
    },
    "bodyPatterns" : [ {
      "matchesJsonPath" : "$[?(@.type =~ /((.*))/)]"
    } ]
  },
  "response" : {
    "status" : 400,
    "body" : "{\"errorCode\":\"BAD_REQUEST\",\"httpCode\":400,\"errors\":[\"400 Bad Request\"]}",
    "headers" : {
      "Content-Type" : "application/hal+json;charset=UTF-8"
    },
    "transformers" : [ "response-template" ]
  },
  "uuid" : "0a72710f-39d8-4beb-be53-358bb5563c57",
  "priority" : 198
}
```

my expectation is when I hit the endpoint `/v1/application` then I should get `400 Bad Request` instead I always get `200 OK` and this is because it matches with request `ANY` loaded as stub, which I have not defined anywhere in my code. Any leads how that part is generated? Could be from my integration test, or from another 3rd party library that I am using as a dependency? 

is it possible somehow to overwrite that stub? I tried to write a stub to overwrite that, with code below, but no luck.

```
import org.springframework.cloud.contract.spec.Contract

Contract.make {
    description "should return load quote result"

    request {
        method 'ANY'
        url('/')
    }

    response {
        status OK()
    }
    priority(100000)
}
```

**Comment by marcingrzejszczak** [comment]
I have no idea where the

```
WireMock                                 : Received request to /mappings with body {
  "id" : "3c192245-23cc-4292-bbd9-2e9275ff5435",
  "request" : {
    "method" : "ANY"
  },
  "response" : {
    "status" : 200
  },
  "uuid" : "3c192245-23cc-4292-bbd9-2e9275ff5435"
}
```

comes from. For sure it doesn't come from Spring Cloud Contract. The ping and health indeed are registered https://github.com/spring-cloud/spring-cloud-contract/blob/master/spring-cloud-contract-stub-runner/src/main/java/org/springframework/cloud/contract/stubrunner/provider/wiremock/WireMockHttpServerStub.java#L265-L268

**Comment by theburhan** [comment]
Thanks for your quick reply, I have defined only 1 stub. Is there a way to log the class path which may explain from where exactly stubs are loaded?

**Comment by marcingrzejszczak** [comment]
Check out http://cloud.spring.io/spring-cloud-contract/spring-cloud-contract.html#_how_can_i_debug_the_requestresponse_being_sent_by_the_generated_tests_client or put a breakpoint in WireMock where it receives a mapping.

**Comment by theburhan** [comment]
I'll give it a try and will leave my feedback here. Thanks.

**Comment by theburhan** [comment]
Ok finally I have found the problem
```
Contract.make {
    request {
        method POST()
        headers {
            contentType 'application/json'
            accept 'application/json'
        }
        url('/v1/application')
        body(file("payload/request/yyy.json"))
        bodyMatchers {
        }
    }
    response {
        status 400
        body(file("payload/response/zzz.json"))
    }
}
```

the content of file yyy.json
```
{

}

```

attempting to map a string json to `StubMapping.class` results in HttpMethod to ANY by default.

https://github.com/tomakehurst/wiremock/blob/master/src/main/java/com/github/tomakehurst/wiremock/stubbing/StubMapping.java#L72
